examples/rust/custom_space_view/src/color_coordinates_space_view.rs

```diff
@@ -0,0 +1,248 @@
+    ///
+    /// In this sample we show a combo box to select the color coordinates mode.
+    fn selection_ui(
+        &self,
+        _ctx: &mut ViewerContext<'_>,
+        ui: &mut egui::Ui,
+        state: &mut Self::State,
+        _space_origin: &EntityPath,
+        _space_view_id: SpaceViewId,
+    ) {
+        ui.horizontal(|ui| {
+            ui.label("Coordinates mode");
+            egui::ComboBox::from_id_source("color_coordinates_mode")
+                .selected_text(state.mode.to_string())
+                .show_ui(ui, |ui| {
+                    ui.style_mut().wrap = Some(false);
+                    ui.set_min_width(64.0);
+
+                    let mut selectable_value = |ui: &mut egui::Ui, e| {
+                        ui.selectable_value(&mut state.mode, e, e.to_string())
+                    };
+                    selectable_value(ui, ColorCoordinatesMode::Hs);
+                    selectable_value(ui, ColorCoordinatesMode::Hv);
+                    selectable_value(ui, ColorCoordinatesMode::Rg);
+                });
+        });
+    }
+
+    /// The contents of the Space View window and all interaction within it.
+    ///
+    /// This is called with a fully populated scene which is a combination of all
+    /// scene contexts, parts and selection highlights.
+    fn ui(
+        &self,
+        ctx: &mut ViewerContext<'_>,
+        ui: &mut egui::Ui,
+        state: &mut Self::State,
+        scene: &mut TypedScene<Self>,
+        _space_origin: &EntityPath,
+        space_view_id: SpaceViewId,
+    ) {
+        egui::Frame {
+            inner_margin: re_ui::ReUi::view_padding().into(),
+            ..egui::Frame::default()
+        }
+        .show(ui, |ui| {
+            let color_at = match state.mode {
+                ColorCoordinatesMode::Hs => |x, y| egui::ecolor::Hsva::new(x, y, 1.0, 1.0).into(),
+                ColorCoordinatesMode::Hv => |x, y| egui::ecolor::Hsva::new(x, 1.0, y, 1.0).into(),
+                ColorCoordinatesMode::Rg => |x, y| egui::ecolor::Rgba::from_rgb(x, y, 0.0).into(),
+            };
+            let position_at = match state.mode {
+                ColorCoordinatesMode::Hs => |c: egui::Color32| {
+                    let hsva: egui::ecolor::Hsva = c.into();
+                    (hsva.h, hsva.s)
+                },
+                ColorCoordinatesMode::Hv => |c: egui::Color32| {
+                    let hsva: egui::ecolor::Hsva = c.into();
+                    (hsva.h, hsva.v)
+                },
+                ColorCoordinatesMode::Rg => |c: egui::Color32| {
+                    let rgba: egui::ecolor::Rgba = c.into();
+                    (rgba.r(), rgba.g())
+                },
+            };
+            draw_color_space(ui, ctx, space_view_id, scene, color_at, position_at);
+        });
+    }
+}
+
+// Draw a mesh for displaying the color space
+// Inspired by https://github.com/emilk/egui/blob/0.22.0/crates/egui/src/widgets/color_picker.rs
+fn draw_color_space(
```

Comment:
egui uses the term `paint` when painting (hence `ui.painter()` below).

but this method does more than just paint, it also allocates the widget, so this should be called something like `color_space_ui`